django/db/models/expressions.py

```diff
@@ -1419,3 +1419,27 @@ class ValueRange(WindowFrame):
 
     def window_frame_start_end(self, connection, start, end):
         return connection.ops.window_frame_range_start_end(start, end)
+
+
+class Collate(Expression):
+    template = "%(expression)s COLLATE %(collation)s"
+
+    def __init__(self, expression, collation):
+        super().__init__(output_field=None)
+        self.expression = expression
+        self.collation = Value(collation)
+
+    def get_source_expressions(self):
+        return [self.expression, self.collation]
+
+    def set_source_expressions(self, exprs):
+        self.expression, self.collation = exprs
```

Comment:
I don't think database allows collation to be an expression thus I shouldn't not be resolved.

```suggestion
    def get_source_expressions(self):
        return [self.expression]

    def set_source_expressions(self, exprs):
        self.expression, = exprs
```